src/adapter/src/coord/sequencer/inner/cluster.rs

```diff
@@ -208,37 +203,43 @@ impl Coordinator {
             }
         }
 
+        match &options.workload_class {
+            Set(wc) => new_config.workload_class.clone_from(wc),
+            Reset => new_config.workload_class = None,
+            Unchanged => {}
+        }
+
         if new_config == config {
             return Ok(StageResult::Response(ExecuteResponse::AlteredObject(
                 ObjectType::Cluster,
             )));
         }
 
-        match (&config.variant, new_config.variant) {
-            (Managed(config), Managed(new_config)) => {
+        match (&config.variant, &new_config.variant) {
+            (Managed(_), Managed(_)) => {
                 self.sequence_alter_cluster_managed_to_managed(
                     Some(session),
                     cluster_id,
-                    config,
                     new_config,
                     ReplicaCreateDropReason::Manual,
                 )
                 .await?;
```

Comment:
Previously `ClusterConfig::variant` was the only field in `ClusterConfig`, so we only passed that to the various `sequence_alter_cluster_*_to_*` methods. Now `ClusterConfig` also contains the `workload_class`, so we need to pass the whole `ClusterConfig`. This is a bit annoying because now the `sequence_alter_cluster_*_to_*` need to assert that `ClusterConfig::variant` is the expected variant, rather than having that ensured by the type system. Not sure if there is a better way to handle this.

Reply:
Seems reasonable to me.